src/java.base/share/classes/java/time/ZonedDateTime.java

```diff
@@ -1301,15 +1301,14 @@ public ZonedDateTime with(TemporalAdjuster adjuster) {
     @Override
     public ZonedDateTime with(TemporalField field, long newValue) {
         if (field instanceof ChronoField chronoField) {
-            switch (chronoField) {
-                case INSTANT_SECONDS:
-                    return create(newValue, getNano(), zone);
-                case OFFSET_SECONDS:
+            return switch (chronoField) {
+                case INSTANT_SECONDS -> create(newValue, getNano(), zone);
+                case OFFSET_SECONDS -> {
                     ZoneOffset offset = ZoneOffset.ofTotalSeconds(chronoField.checkValidIntValue(newValue));
-                    return resolveOffset(offset);
-            }
-            return resolveLocal(dateTime.with(field, newValue));
-        }
+                    yield resolveOffset(offset);
+                }
+                default -> resolveLocal(dateTime.with(field, newValue));
+        };
```

Comment:
It looks like that the closing braces `}` is missing here.

Reply:
Hi Aleksei, thanks for spotting this. I've added that in now. See 5fa4e3c